osbenchmark/workload_generator/workload_generator.py

```diff
@@ -41,10 +41,24 @@ def process_template(templates_path, template_filename, template_vars, output_pa
     with open(output_path, "w") as f:
         f.write(template.render(template_vars))
 
+def check_if_indices_and_total_docs_match(indices, total_docs, docs_were_requested):
+    '''
+    if --total-docs were provided, the input length needs to be match indices.
 
-def extract_mappings_and_corpora(client, output_path, indices_to_extract):
+    Valid examples: --indices=movies,actors --total-docs=1000,2000
+    Invalid examples: --indices=movies,actors --total-docs=1000 OR
+    --indices=movies --total-docs=1000,2000
+    '''
+    if docs_were_requested and len(indices) != len(total_docs):
+        raise exceptions.SystemSetupError("Both --indices and --total-docs were provided but missing corresponding indices or total docs.")
+
+def extract_mappings_and_corpora(client, output_path, indices_to_extract, total_docs_requested):
     indices = []
     corpora = []
+    docs_were_requested = total_docs_requested is not None and len(total_docs_requested) > 0
+
+    check_if_indices_and_total_docs_match(indices_to_extract, total_docs_requested, docs_were_requested)
+
```

Comment:
Should likely work just with:
```
     check_if_indices_and_total_docs_match(indices_to_extract, total_docs_requested, number_of_docs_requested)
```

Reply:
The function `check_if_indices_and_total_docs_match()` takes in a list of indices to extract docs from, a list of doc count to take from each index, and a boolean of whether docs were requested. I included the last parameter `docs_were_requested` since it's used in different places in the method and would reduce the amount of checks we have to rewrite.